### PR TITLE
can find an click links with regex href values

### PR DESCRIPTION
**Pull request by YSavir**
Adds regex support when searching for links with `href` filters.

Solves #1510.

### CODE REVIEW

File: lib/capybara/query.rb [hunk 1]
```diff
@@ -43,6 +43,10 @@ def matches_filters?(node)
         regexp = options[:text].is_a?(Regexp) ? options[:text] : Regexp.escape(options[:text].to_s)
         return false if not node.text(visible).match(regexp)
       end
+      if options[:href]
```
Comment (twalpole) [inline lib/capybara/query.rb]: This section is for filters that are common to all queries/selectors  -- href is not so it shouldn't be here, it should be in the relevant selectors

File: lib/capybara/selector.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+require 'pry'
```
Comment (twalpole) [inline lib/capybara/selector.rb]: non-necessary change

File: lib/capybara/query.rb (hunk 1 above)
Comment (YSavir) [inline lib/capybara/query.rb]: Will change.

File: lib/capybara/selector.rb (hunk 2 above)
Comment (YSavir) [inline lib/capybara/selector.rb]: Woops. Missed that in the `git diff` :/.

**Comment by twalpole** [comment]
@YSavir Other than the few line comments I made, its looking good -- please also update the has_link and click_link documentation to show they can take a regex for :href

File: lib/capybara/query.rb (hunk 1 above)
Comment (YSavir) [inline lib/capybara/query.rb]: Turns out that the changes I made to the `link` selector made this filter unnecessary anyway.

**Comment by YSavir** [comment]
Added changes. Should I squash the commits?

File: lib/capybara/node/actions.rb [hunk 3]
```diff
@@ -21,7 +21,7 @@ def click_link_or_button(locator, options={})
       #
       # @param [String] locator         text, id, title or nested image's alt attribute
       # @param options
-      # @option options [String] :href    The value the href attribute must equal
+      # @option options [String] :href    The value the href attribute must equal. May be either a string or regular expression.
```
Comment (twalpole) [inline lib/capybara/node/actions.rb]: This should be [String, Regexp]

File: lib/capybara/node/matchers.rb [hunk 4]
```diff
@@ -220,7 +220,7 @@ def has_no_css?(path, options={})
       #
       # @param [String] locator           The text or id of a link to check for
       # @param options
-      # @option options [String] :href    The value the href attribute must be
+      # @option options [String] :href    The value the href attribute must be. May be either a string or regular expression.
```
Comment (twalpole) [inline lib/capybara/node/matchers.rb]: [String, Regexp]

**Comment by twalpole** [comment]
Once the documentation fixes are made, yes please squash

**Comment by YSavir** [comment]
Done and done.

**Comment by twalpole** [comment]
Thanks

**Pull request merged**